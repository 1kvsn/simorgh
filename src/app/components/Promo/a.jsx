import React from 'react';
import styled from '@emotion/styled';
import { C_EBON } from '@bbc/psammead-styles/colours';

const StyledA = styled.a`
  color: ${C_EBON};
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
  &:before {
    position: absolute;
    bottom: 0;
    top: 0;
    left: 0;
    right: 0;
    content: '';
  }
`;

const A = props => {
  return <StyledA {...props} />;
};

A.propTypes = {};

A.defaultProps = {};

export default A;
