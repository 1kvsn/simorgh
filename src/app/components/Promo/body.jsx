import styled from '@emotion/styled';

import { getSansRegular } from '@bbc/psammead-styles/font-styles';
import { getBodyCopy } from '#legacy/gel-foundations/src/typography';
import { C_EBON } from '@bbc/psammead-styles/colours';

const P = styled.p`
  ${({ service }) => getSansRegular(service)}
  ${({ script }) => getBodyCopy(script)}
  color: ${C_EBON}
`;

export default P;
