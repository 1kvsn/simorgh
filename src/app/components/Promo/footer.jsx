import React from 'react';
import styled from '@emotion/styled';

import { getSansRegular } from '@bbc/psammead-styles/font-styles';
import { getBrevier } from '@bbc/gel-foundations/typography';
import { C_RHINO } from '@bbc/psammead-styles/colours';

const StyledFooter = styled.footer`
  ${({ service }) => getSansRegular(service)}
  ${({ script }) => getBrevier(script)}
  color: ${C_RHINO}
`;

const Footer = props => {
  return <StyledFooter {...props} />;
};

Footer.propTypes = {};

Footer.defaultProps = {};

export default Footer;
