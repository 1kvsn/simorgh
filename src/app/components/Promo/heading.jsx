import React from 'react';
import styled from '@emotion/styled';

import { getSansRegular } from '@bbc/psammead-styles/font-styles';
import { getBodyCopy } from '@bbc/gel-foundations/typography';
import { C_EBON } from '@bbc/psammead-styles/colours';

const H1 = styled.h1`
  ${({ service }) => getSansRegular(service)}
  ${({ script }) => getBodyCopy(script)}
  color: ${C_EBON}
`;

const Heading = props => {
  return <H1 {...props} />;
};

Heading.propTypes = {};

Heading.defaultProps = {};

export default Heading;
