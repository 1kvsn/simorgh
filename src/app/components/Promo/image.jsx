import React from 'react';
import styled from '@emotion/styled';
import ImagePlaceholder from '@bbc/psammead-image-placeholder';
import { string } from 'prop-types';

const Img = styled.img`
  width: 100%;
`;

const Image = props => {
  return (
    <ImagePlaceholder ratio={56.25}>
      <Img {...props} />
    </ImagePlaceholder>
  );
};

Image.propTypes = {
  alt: string.isRequired,
};

Image.defaultProps = {};

export default Image;
