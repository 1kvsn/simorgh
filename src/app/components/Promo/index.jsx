import React, { useContext } from 'react';
import styled from '@emotion/styled';
import { arrayOf, element } from 'prop-types';
import { ServiceContext } from '#contexts/ServiceContext';

import Image from './image';
import Heading from './heading';
import Body from './body';
import Footer from './footer';
import A from './a';

const PromoContext = React.createContext({});
const withPromoContext = Component => props =>
  (
    <PromoContext.Consumer>
      {context => <Component {...context} {...props} />}
    </PromoContext.Consumer>
  );

const Wrapper = styled.section`
  position: relative;
`;

const Promo = ({ children }) => {
  const { script, service } = useContext(ServiceContext);
  return (
    <Wrapper>
      <PromoContext.Provider value={{ script, service }}>
        {children}
      </PromoContext.Provider>
    </Wrapper>
  );
};

Promo.Image = withPromoContext(Image);
Promo.Heading = withPromoContext(Heading);
Promo.Body = withPromoContext(Body);
Promo.Footer = withPromoContext(Footer);
Promo.A = withPromoContext(A);

Promo.propTypes = {
  children: arrayOf(element).isRequired,
};

Promo.defaultProps = {};

export default Promo;
