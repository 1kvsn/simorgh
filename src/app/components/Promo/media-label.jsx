import React from 'react';
import moment from 'moment-timezone';
import styled from '@emotion/styled';
import { shape, string, number, oneOf } from 'prop-types';
import { scriptPropType } from '@bbc/gel-foundations/prop-types';
import { GEL_SPACING_HLF } from '@bbc/gel-foundations/spacings';
import { C_WHITE, C_EBON } from '@bbc/psammead-styles/colours';
import { getMinion } from '@bbc/gel-foundations/typography';
import { getSansRegular } from '@bbc/psammead-styles/font-styles';
import { mediaIcons } from '@bbc/psammead-assets/svgs';
import formatDuration from '#lib/utilities/formatDuration';

export const TYPES = {
  VIDEO: 'video',
  AUDIO: 'audio',
  PHOTO_GALLERY: 'photogallery',
};

const StyledMediaIndicator = styled.div`
  padding: ${GEL_SPACING_HLF};
  color: ${C_EBON};
  background-color: ${C_WHITE};
  ${({ service }) => getSansRegular(service)}
  ${({ script }) => script && getMinion(script)};
`;

const StyledTime = styled.time`
  padding: ${GEL_SPACING_HLF};
  position: relative;
  top: 0.09rem;
`;

const formatChildren = children => {
  if (!children) return null;
  const duration = moment.duration(children, 'seconds');
  const durationString = formatDuration({ duration });
  const isoDuration = duration.toISOString();
  return <StyledTime dateTime={isoDuration}>{durationString}</StyledTime>;
};

const MediaLabel = ({ script, service, children, type }) => {
  if (!type) return null;
  return (
    <StyledMediaIndicator script={script} service={service}>
      {mediaIcons[TYPES[type]]}
      {formatChildren(children)}
    </StyledMediaIndicator>
  );
};

MediaLabel.propTypes = {
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
  type: oneOf(Object.keys(TYPES)),
  children: number,
};

MediaLabel.defaultProps = {
  children: null,
  type: null,
};

export default MediaLabel;
