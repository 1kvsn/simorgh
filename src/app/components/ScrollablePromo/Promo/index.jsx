import React, { useContext } from 'react';
import { object } from 'prop-types';
import styled from '@emotion/styled';
import pathOr from 'ramda/src/pathOr';
import { getPica } from '@bbc/gel-foundations/dist/typography';
import { getSerifRegular } from '@bbc/psammead-styles/dist/font-styles';
import { C_GREY_6 } from '@bbc/psammead-styles/dist/colours';
import {
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
} from '@bbc/gel-foundations/dist/spacings';
import {
  GEL_GROUP_0_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/dist/breakpoints';
import { ServiceContext } from '#contexts/ServiceContext';
import filterForBlockType from '#lib/utilities/blockHandlers';

const C_GREY_8 = '#202224';

const Link = styled.a`
  ${({ script }) => script && getPica(script)}
  ${({ service }) => service && getSerifRegular(service)}
  width: 100%;

  text-overflow: ellipsis;
  text-decoration: none;

  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  color: ${C_GREY_8};
  &:visited {
    color: ${C_GREY_6};
  }
`;

const PromoBox = styled.div`
  margin-bottom: ${GEL_SPACING_TRPL};
  background-color: #ffffff;
  padding: ${GEL_SPACING_DBL};
  @media (min-width: ${GEL_GROUP_0_SCREEN_WIDTH_MIN}) {
    width: 237px;
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    width: 178px;
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    width: 205px;
  }
`;

const Promo = ({ block }) => {
  const { script, service, dir } = useContext(ServiceContext);
  const textBlock = filterForBlockType(
    pathOr({}, ['model', 'blocks'], block),
    'text',
  );
  const href = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'locator'],
    textBlock,
  );
  const title = pathOr(
    '',
    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'text'],
    textBlock,
  );
  return (
    <PromoBox dir={dir}>
      <Link href={href} service={service} script={script} dir={dir}>
        {title}
      </Link>
    </PromoBox>
  );
};

Promo.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  block: object.isRequired,
};

export default Promo;
