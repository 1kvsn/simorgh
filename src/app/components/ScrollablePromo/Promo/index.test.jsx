import React from 'react';
import { render } from '@testing-library/react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { PromoSingleBlock } from '../helpers/fixtureData';
import Promo from '.';

describe('ScrollablePromo', () => {
  it('should render a link', () => {
    const { queryByRole } = render(<Promo block={PromoSingleBlock} />);
    expect(queryByRole('link')).toBeInTheDocument();
  });

  shouldMatchSnapshot(
    'should truncate the link text if more than 3 lines',
    <Promo block={PromoSingleBlock} />,
  );
});
