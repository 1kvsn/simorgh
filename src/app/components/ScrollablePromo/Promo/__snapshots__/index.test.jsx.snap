// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollablePromo should truncate the link text if more than 3 lines 1`] = `
.emotion-0 {
  margin-bottom: 1.5rem;
  background-color: #ffffff;
  padding: 1rem;
}

@media (min-width: 0rem) {
  .emotion-0 {
    width: 237px;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    width: 178px;
  }
}

@media (min-width: 63rem) {
  .emotion-0 {
    width: 205px;
  }
}

.emotion-2 {
  width: 100%;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  color: #202224;
}

.emotion-2:hover,
.emotion-2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:visited {
  color: #545658;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <a
      class="emotion-2 emotion-3"
      href="https://www.bbc.com/mundo"
    >
      This is a very long headline. I am creating this for a test purpose. I love creating these type of tests. I really do not know what to write.
    </a>
  </div>
</div>
`;
